src/librustc/ty/mod.rs

```diff
@@ -606,9 +680,10 @@ impl<'a, T> IntoIterator for &'a Slice<T> {
 impl<'tcx> serialize::UseSpecializedDecodable for &'tcx Slice<Ty<'tcx>> {}
 
 impl<T> Slice<T> {
+    #[inline(always)]
     pub fn empty<'a>() -> &'a Slice<T> {
         unsafe {
-            mem::transmute(slice::from_raw_parts(0x1 as *const T, 0))
+            &*(EMPTY_SLICE as *const _)
```

Comment:
I'm wondering if this can trigger some kind of undefined behavior? Maybe `mem::transmute(EMPTY_SLICE)` would be better.

Reply:
I can get rid of the optimization for the empty slice and see if that helps...

Reply:
Removing it does make things work.